Add alma10 sysroot

h-vetinari opened the pull request:
https://repo.almalinux.org/vault/10.0/BaseOS/ exists but is still empty. It should be filled once alma 10.1 is out, which might happen in November. In the meantime, use packages from https://repo.almalinux.org/almalinux. Later, we should simply be able to revert 0de3517704e26ea237c592cd8b58eedadb0e5acd.

Fixes #91

h-vetinari committed:
adapt update.py for alma10

h-vetinari committed:
vault is not populated yet

h-vetinari committed:
no more glibc-headers in alma10

h-vetinari committed:
regenerate meta.yaml

h-vetinari committed:
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.17.06.41.22

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ License exception is not an SPDX exception.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17803669240](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17803669240). Examine the logs at this URL for more detail.</sub>

h-vetinari committed:
MNT: Re-rendered with conda-smithy 3.53.0 and conda-forge-pinning 2025.11.09.23.10.12

Other tools:
- conda-build 25.9.0
- rattler-build 0.49.0
- rattler-build-conda-compat 1.4.9

h-vetinari committed:
move missing_dso_whitelist to respective output

h-vetinari committed:
packaging now reflects `/usr`-merge

h-vetinari commented:
Good that I added some extra tests; one of them caught
```
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/lib is a symlink with no target
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/usr/bin/ld.so is a symlink with no target
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib64/libBrokenLocale.so is a symlink with no target
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib64/libanl.so is a symlink with no target
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib64/libmvec.so is a symlink with no target
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib64/libresolv.so is a symlink with no target
$SP_DIR/conda_build/build.py:1537: UserWarning: file $PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib64/libthread_db.so is a symlink with no target
```
(or would have caught it, had the test not failed otherwise 😅)

h-vetinari committed:
use newest image

h-vetinari committed:
MNT: Re-rendered with conda-smithy 3.53.1 and conda-forge-pinning 2025.11.15.23.15.53

Other tools:
- conda-build 25.9.0
- rattler-build 0.49.0
- rattler-build-conda-compat 1.4.9

h-vetinari committed:
fail on error

beckermr reviewed (dismissed):
We have at least historrically kept branches on the upstream feedstock for each sysroot version. We should do that here and so the PR needs to be redirect to a branch. Feel free to make one.

h-vetinari commented:
> We have at least historically kept branches on the upstream feedstock for each sysroot version.

Look at the branches here

<img width="525" height="635" alt="image" src="https://github.com/user-attachments/assets/00b7d759-0ff8-4ce8-b6ab-f65a1c43c070" />

These correspond to

| branch | Distro | glibc version |
|--------|--------|--------|
| `v2.12` | CentOS 6 | 2.12 |
| `v2.17` | CentOS 7 | 2.17 |
| `v2.28` | Alma 8 | 2.28 |
| `main` | Alma 9 | 2.34 | 

In other words, already now the newest version doesn't have a special branch. This is regular feedstock behaviour if there are several maintenance branches.

> We should do that here and so the PR needs to be redirect to a branch. Feel free to make one.

What we should do here is to merge this PR to `main` as usual, and once that happens, create a `v2.34` branch that matches the state before this PR (i.e. the main branch as of right now). I'm happy to create that branch once this is in.

h-vetinari committed:
Ensure paths without `/usr` keeps working; expand tests

h-vetinari committed:
also keep `/bin/*` & `/sbin/*` working

h-vetinari committed:
remove s390x workaround

h-vetinari commented:
OK, this should be ready now! PTAL :)

h-vetinari commented:
Branch `v2.34` is there too now: https://github.com/conda-forge/linux-sysroot-feedstock/tree/v2.34

beckermr commented on recipe/meta.yaml:
```diff
@@ -16,65 +16,57 @@ source:
-    sha256: 50808a6a71fd9a3d5b3db2604f825af8aa6cf0020dd8f1d0edca1b0bc77729f9  # [cross_target_platform == "linux-ppc64le"]
-    sha256: e7a9df0d9eb747b1dacf13636a4defb42123a0aa3d166adff9970736d1be073c  # [cross_target_platform == "linux-s390x"]
-
-  - folder: binary-glibc-headers                                              # [cross_target_platform in ("linux-64", "linux-s390x")]
```
I'd be nice if they stopped moving stuff around.

beckermr reviewed (approved):
Thank you!

luhenry commented:
AlmaLinux 10 Kitten is available on RISC-V: https://almalinux.org/blog/2026-03-17-almalinux-goes-riscv/. Does that require updating?
